src/QSymbolicsBase/predefined.jl

```diff
@@ -207,25 +241,61 @@ function Base.show(io::IO, x::SProjector)
     print(io,"]")
 end
 
-"""Dagger a Ket into Bra."""
-@withmetadata struct SDagger <: Symbolic{AbstractBra}
-    ket::Symbolic{AbstractKet}
+"""Dagger, i.e., adjoint of quantum objects (kets, bras, operators)
+
+```jldoctest 
+julia> a = SKet(:a, SpinBasis(1//2)); A = SOperator(:A, SpinBasis(1//2));
+
+julia> dagger(2*im*A*a)
+0 - 2im⟨a|A†
+
+julia> B = SOperator(:B, SpinBasis(1//2));
+
+julia> dagger(A*B)
+B†A†
+
+julia> dagger(hermitian(A))
+A 
+
+julia> dagger(unitary(A)) 
+A⁻¹
+```
+"""
+@withmetadata struct SDagger{T<:QObj} <: Symbolic{T}
+    obj
 end
 isexpr(::SDagger) = true
 iscall(::SDagger) = true
-arguments(x::SDagger) = [x.ket]
+arguments(x::SDagger) = [x.obj]
 operation(x::SDagger) = dagger
 head(x::SDagger) = :dagger
-children(x::SDagger) = [:dagger, x.ket]
-dagger(x::Symbolic{AbstractKet}) = SDagger(x)
-dagger(x::SScaledKet) = SScaledBra(x.coeff, dagger(x.obj))
+children(x::SDagger) = [:dagger, x.obj]
```

Comment:
With the switch to the newer version of the foundational dependencies, do we need both `arguments` and `children` or only one of them?

Reply:
According to most recent `TermInterface` docs, when `isexpr` is true, both `head` and `children` must be defined. Similarly, `operations` and `arguments` must be defined when `iscall` is true. If `iscall` is true, then `isexpr` must be true.

Reply:
So to answer your question, for most cases, yes.